source/extensions/transport_sockets/alts/noop_transport_socket_callbacks.h

```diff
@@ -0,0 +1,34 @@
+#include "envoy/network/transport_socket.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace TransportSockets {
+namespace Alts {
+
+/**
+ * A TransportSocketCallbacks for wrapped TransportSocket object. Some
+ * TransportSocket implementation wraps another socket which does actual I/O.
+ * This class is used by the wrapped socket as its callbacks instead of the real
+ * connection to hold back callbacks from the underlying socket to connection.
+ */
+class NoOpTransportSocketCallbacks : public Network::TransportSocketCallbacks {
+public:
+  explicit NoOpTransportSocketCallbacks(Network::TransportSocket* parent) : parent_(parent) {}
```

Comment:
All you need is parent_->callbacks(), so why just not taking a `Network::TransportSocketCallbacks&` as parameter?

Reply:
It depends on the timing when NoOpTransportSocketCallbacks is initialized and parent socket's setTransportSocketCallbacks() is called. In #4153 it is initialized before parent knows its callbacks.  It can take a TransportSocketCallbacks** too, but I feel not much more simpler.